Doesn't compile on Mac / clang

Some errors here and elsewhere from ce01e005 I think.
https://github.com/OnlineCop/kq-fork/blob/76d3fe1b068d8fd9ccf8b7670b47ccbe96361476/src/tiledmap.cpp#L682

```
kq-fork/src/tiledmap.cpp:682:58: error: expected
      '(' for function-style cast or type construction
                unsigned int index = std::clamp(unsigned int(i + zone.x ...
                                                ~~~~~~~~ ^
1 warning generated.
kq-fork/src/entity.cpp:546:47: error: expected '(' for
      function-style cast or type construction
        int dest_tile_x = std::clamp(unsigned int(source_tile + dx), 0U...
                                     ~~~~~~~~ ^
kq-fork/src/entity.cpp:547:47: error: expected '(' for
      function-style cast or type construction
        int dest_tile_y = std::clamp(unsigned int(source_tile + dy * g_m...
                                     ~~~~~~~~ ^
1 error generated.
```

Also, (sorry to be the bearer of bad news)
```
Undefined symbols for architecture x86_64:
  "KZone::ZONE_NONE", referenced from:
      tmx_map::set_current() in tiledmap.cpp.o
ld: symbol(s) not found for architecture x86_64
```

missing #include zone.h ? No, actually I don't know what's up with this

--- a/include/gfx.h
+++ b/include/gfx.h
@@ -25,7 +25,7 @@ class Raster
     void fill(uint8_t colour);
     uint8_t& ptr(int16_t x, int16_t y)
     {
-        return data[x + y * size_t(stride)];
+        return data[x + y * static_cast<size_t>(stride)];
     }
     const uint16_t width, height;
     const uint16_t stride;

--- a/include/zone.h
+++ b/include/zone.h
@@ -4,7 +4,7 @@ class KZone
 {
   public:
     // A zone value of '0' may trigger events on maps where s_map::zero_zone or tmx_map::zero_zone is non-zero.
-    static const int ZONE_NONE = 0;
+    static constexpr int ZONE_NONE = 0;
 
     int x;
     int y;

--- a/src/entity.cpp
+++ b/src/entity.cpp
@@ -543,8 +543,8 @@ static int move(t_entity target_entity, int dx, int dy)
     if (dx && dy)
     {
         source_tile = tile_y * g_map.xsize + tile_x;
-        int dest_tile_x = std::clamp(unsigned int(source_tile + dx), 0U, g_map.xsize * g_map.ysize - 1);
-        int dest_tile_y = std::clamp(unsigned int(source_tile + dy * g_map.xsize), 0U, g_map.xsize * g_map.ysize - 1);
+        int dest_tile_x = std::clamp(static_cast<unsigned int>(source_tile + dx), 0U, g_map.xsize * g_map.ysize - 1);
+        int dest_tile_y = std::clamp(static_cast<unsigned int>(source_tile + dy * g_map.xsize), 0U, g_map.xsize * g_map.ysize - 1);
         if (Game.Map.zone_array[source_tile] != Game.Map.zone_array[dest_tile_x] ||
             Game.Map.zone_array[source_tile] != Game.Map.zone_array[dest_tile_y])
         {

--- a/src/gfx.cpp
+++ b/src/gfx.cpp
@@ -33,10 +33,10 @@ Raster::Raster(uint16_t w, uint16_t h)
 void Raster::blitTo(Raster* target, int16_t src_x, int16_t src_y, uint16_t src_w, uint16_t src_h, int16_t dest_x,
                     int16_t dest_y, uint16_t dest_w, uint16_t dest_h, bool masked)
 {
-    auto x0 = std::max(0, int(dest_x));
-    auto x1 = std::min(int(target->width), dest_x + dest_w);
-    auto y0 = std::max(0, int(dest_y));
-    auto y1 = std::min(int(target->height), dest_y + dest_h);
+    auto x0 = std::max(0, static_cast<int>(dest_x));
+    auto x1 = std::min(static_cast<int>(target->width), dest_x + dest_w);
+    auto y0 = std::max(0, static_cast<int>(dest_y));
+    auto y1 = std::min(static_cast<int>(target->height), dest_y + dest_h);
     bool stretch = (dest_w != src_w) || (dest_h != src_h);
     // Four cases - stretch + masked, stretch + not masked, not stretch + masked, not stretch + not masked.
     if (stretch)

--- a/src/kq.cpp
+++ b/src/kq.cpp
@@ -1014,10 +1014,9 @@ void KGame::prepare_map(int msx, int msy, int mvx, int mvy)
 {
     Raster* pcxb;
     unsigned int i;
-    size_t mapsize;
     unsigned int o;
 
-    mapsize = (size_t)g_map.xsize * (size_t)g_map.ysize;
+    size_t mapsize = g_map.xsize * g_map.ysize;
 
     draw_background = draw_middle = draw_foreground = draw_shadow = 0;
 

--- a/src/music.cpp
+++ b/src/music.cpp
@@ -102,7 +102,7 @@ void KMusic::set_music_volume(int volume)
         return;
     }
     mvol = float(std::clamp(volume, 0, 250)) / 250.0f;
-    Mix_VolumeMusic(int(dvol * mvol * float(MIX_MAX_VOLUME)));
+    Mix_VolumeMusic(static_cast<int>(dvol * mvol * float(MIX_MAX_VOLUME)));
 }
 
 /*! \brief Poll the music  (SDL2_Mixer)
@@ -212,8 +212,8 @@ void KMusic::set_volume(int sound_volume)
         return;
     }
     dvol = float(std::clamp(sound_volume, 0, 250)) / 250.0f;
-    Mix_Volume(-1, int(128.0f * dvol));
-    Mix_VolumeMusic(int(dvol * mvol * float(MIX_MAX_VOLUME)));
+    Mix_Volume(-1, static_cast<int>(128.0f * dvol));
+    Mix_VolumeMusic(static_cast<int>(dvol * mvol * float(MIX_MAX_VOLUME)));
 }
 KMusic Music;
 

--- a/src/tiledmap.cpp
+++ b/src/tiledmap.cpp
@@ -679,7 +679,7 @@ void tmx_map::set_current()
         {
             for (int j = 0; j < zone.h; ++j)
             {
-                unsigned int index = std::clamp(unsigned int(i + zone.x + xsize * (j + zone.y)), 0U, g_map.xsize * g_map.ysize - 1);
+                unsigned int index = std::clamp(static_cast<unsigned int>(i + zone.x + xsize * (j + zone.y)), 0U, g_map.xsize * g_map.ysize - 1);
                 Game.Map.zone_array[index] = zone.n;
             }
         }

